frontend/hub/dashboard/CollectionCard.tsx

```diff
@@ -35,7 +37,13 @@ export function CollectionCard(props: { collection: CollectionVersionSearch }) {
         itemToCardFn={(item: CollectionVersionSearch) => ({
           id: item.collection_version.name,
           icon: <AnsibleTowerIcon />, // TODO: Update logo to use avatar_url if it exists
-          title: <TextCell text={item.collection_version.name} />,
+          title: (
+            <TextCell
+              text={item.collection_version.name}
+              to={RouteObj.CollectionDetails.replace(':id', item.collection_version.name)}
```

Comment:
not sure if you want to be fixing this in the same PR, but:

a name is not enough to identify a collection by,

any urls to collection detail will need to include the repository, namespace and collection names, and optionally the version

Reply:
Something like that would make sense, yes :+1: 

Except .. when `version` is not set, you'd want to use `is_highest=true` as well, otherwise you'll get all the versions, instead of 1 per collection.

---

What current hub actually does on that screen is:

```
# what you already have without version but with is_highest
/api/automation-hub/v3/plugin/ansible/search/collection-versions
?repository_name=published&namespace=ansible&name=network&is_highest=true&offset=0&limit=10

# to populate the version selector (it opens a separate modal when count>10 but uses this to show a dropdown when <=10, and to drive the decision which to open)
/api/automation-hub/v3/plugin/ansible/search/collection-versions
?repository_name=published&namespace=ansible&name=network&order_by=-version&offset=0&limit=10

# for "content" - README, documentation tab
/api/automation-hub/pulp/api/v3/content/ansible/collection_versions
?namespace=ansible&name=network&version=2.0.0&offset=0&limit=10

# dependencies for the "can delete" decision
/api/automation-hub/v3/plugin/ansible/search/collection-versions
?dependency=ansible.network&offset=0&limit=10

# namespace ownership implies change permissions
/api/automation-hub/_ui/v1/namespaces/ansible
?include_related=my_permissions
```

And once download count support is merged, this will also grow by...

```
# load repository..
/api/automation-hub/pulp/api/v3/repositories/ansible/ansible/
?name=published&offset=0&limit=1

# load distribution for repository (falls back to `?repository=href&order_by=-pulp_created&limit=1`)
/api/automation-hub/pulp/api/v3/distributions/ansible/ansible/
?name=published&offset=0&limit=1

# load collection data (as opposed to collection version) - for download counts
/api/automation-hub/v3/plugin/ansible/content/published/collections/index/ansible/network/
(no params)
```

(this is handed by `loadCollection` in the old ui - 
https://github.com/ansible/ansible-hub-ui/blob/189a6119e667f0c0ef71563f018c5e3baeb6b171/src/containers/collection-detail/base.ts)

Reply:
Thanks for the additional details. We'll keep this in mind for the future updates to CollectionDetails 👍